basecoin/src/bin/basecoin.rs

```diff
@@ -40,5 +51,69 @@ async fn main() {
             };
             let _ = write!(std::io::stdout(), "{:#?}", query_res);
         }
+        Commands::Tx(c) => match c {
+            TxCmd::Recover(cmd) => {
+                let RecoverCmd {
+                    subject_client_id,
+                    substitute_client_id,
+                } = cmd;
+
+                let subject_client_id =
+                    ClientId::from_str(subject_client_id).expect("valid client ID");
+                let substitute_client_id =
+                    ClientId::from_str(substitute_client_id).expect("valid client ID");
+
+                let hdpath = StandardHDPath::from_str(DEFAULT_DERIVATION_PATH).unwrap();
+
+                let key_pair = match KeyPair::from_seed_file(SEED_FILE_PATH, &hdpath) {
+                    Ok(key_pair) => key_pair,
+                    Err(e) => {
+                        tracing::error!("{e}");
+                        std::process::exit(1);
+                    }
+                };
+
+                let signer = Signer::from(key_pair.account.clone());
+
+                // Create the MsgRecoverClient
+                let msg = MsgRecoverClient {
+                    subject_client_id,
+                    substitute_client_id,
+                    signer,
+                };
+
+                let chain_id = dummy_chain_id();
+                let grpc_addr = cfg.cometbft.grpc_addr.clone();
+                let rpc_addr = cfg.cometbft.rpc_addr.clone();
```

Comment:
```suggestion
                let grpc_addr = format!("http://{}:{}", cfg.server.host, cfg.server.grpc_port)
                    .parse()
                    .expect("valid grpc endpoint");
```

:sweat_smile: it was using the CometBFT grpc endpoint.